### ISSUE TITLE
Unclear Error Message for Deprecated TensorFlow Models

### ISSUE THREAD

**Issue by ajross4** [QATCH-Technologies/nanovisQ#245]
# Bug Report: Unclear Error Message for Deprecated TensorFlow Models

## Summary
When selecting a deprecated TensorFlow model ZIP file on the Evaluate tab (tab 5), the error message "Failed to load model: 'src/' folder missing in archive" is vague and doesn't explain the actual problem. The message should clearly indicate that the old TensorFlow models are incompatible with the current PyTorch-based system.

## Description
Legacy VisQAI model ZIP files that contain now-deprecated TensorFlow models cannot be loaded in the current PyTorch-driven implementation. When users attempt to select these old model files, they receive a cryptic error about a missing 'src/' folder, which doesn't communicate the real issue: the model format is obsolete and no longer supported. Users need clear guidance that they should use the base model and retrain.

## Steps to Reproduce
1. Navigate to Evaluate tab (tab 5)
2. Click to select a model ZIP file
3. Select an old VisQAI model ZIP file containing deprecated TensorFlow models
4. Observe error popup

## Current Error Message
```
Failed to load model: 'src/' folder missing in archive
```

### Problems with Current Message:
- Suggests a file corruption or packaging issue
- Doesn't explain that TensorFlow models are deprecated
- Doesn't mention PyTorch migration
- No guidance on what user should do
- Users may waste time trying to "fix" the ZIP file
- Technical error message rather than user-friendly explanation

## Expected Error Message
The error message should be clear, informative, and actionable:

### Proposed Message:
```
Incompatible Model Format

The selected model file uses a deprecated TensorFlow format that is no longer supported.

Current version requires PyTorch-based models.

Action Required:
• Revert to using the base model
• Retrain your model with the current version

Would you like to load the base model now?
[Load Base Model]  [Cancel]
```

### Alternative (More Concise):
```
Model Not Compatible

This model file is from an older version and cannot be used.

Please use the base model and retrain.

[OK]
```

## Actual Behavior
- Vague error: "'src/' folder missing in archive"
- User confused about what went wrong
- No indication models are deprecated
- No guidance on resolution
- User may attempt to "repair" ZIP file

## Expected Behavior
- Clear message: Model format is deprecated/incompatible
- Explanation: TensorFlow → PyTorch migration
- Actionable guidance: Use base model and retrain
- Optional: Offer to auto-load base model
- User understands the issue immediately

## Impact
- **Severity**: Low-Medium - Error handling/UX issue
- **Priority**: Medium - Affects users with legacy models
- **User Impact**: 
  - Confusion about error cause
  - Time wasted troubleshooting wrong issue
  - Potential support requests
  - Frustration with unclear messaging
  - May not realize models need retraining

## Root Cause
Error detection correctly identifies incompatible model structure but reports technical implementation detail (missing 'src/' folder) rather than user-facing reason (deprecated TensorFlow format).

## Acceptance Criteria
- [ ] Clear error message when deprecated TensorFlow models selected
- [ ] Message explicitly states model format is deprecated/incompatible
- [ ] Message mentions TensorFlow → PyTorch migration (or just "older version")
- [ ] Actionable guidance provided: "use base model and retrain"
- [ ] Technical jargon minimized (or optional: offer to load base model automatically)
- [ ] No more vague "'src/' folder missing" errors for this case
- [ ] User understands what happened and what to do
- [ ] Consistent messaging across all model loading locations

### Do Include:
✓ Clear problem statement ("incompatible", "deprecated format")
✓ Brief reason ("older version", "TensorFlow → PyTorch")
✓ Actionable solution ("use base model and retrain")
✓ User-friendly language

### Don't Include:
✗ Technical implementation details ("'src/' folder missing")
✗ Internal structure requirements
✗ Confusing technical jargon
✗ Details that don't help user resolve issue

## Testing Requirements
- [ ] Test with actual deprecated TensorFlow model ZIP files
- [ ] Verify new error message displays correctly
- [ ] Test that current PyTorch models still load normally
- [ ] Test error message on different platforms (Windows/Linux/macOS)
- [ ] Verify dialog layout and readability
- [ ] Test optional "Load Base Model" button (if implemented)
- [ ] Ensure error doesn't crash or cause other issues

### Before (Poor UX):
1. User selects old TensorFlow model
2. Error: "'src/' folder missing in archive"
3. User confused: "Is my file corrupted?"
4. User tries to fix ZIP file (wrong approach)
5. User contacts support

### After (Good UX):
1. User selects old TensorFlow model
2. Clear message: "Model format deprecated, use base model and retrain"
3. User understands: "I need to retrain"
4. User loads base model and retrains
5. User successful

## Related Components
- Model loading system (Evaluate tab, tab 5)
- ZIP file validation
- Error handling and messaging
- Model format detection
- Base model loading

## Environment Details
- **Location**: Evaluate tab (tab 5)
- **File Type**: VisQAI model ZIP files
- **Old Format**: TensorFlow-based models (deprecated)
- **Current Format**: PyTorch-based models
- **Error**: Model compatibility check

## Migration Context
- Application migrated from TensorFlow to PyTorch
- Old model files no longer compatible
- Users need to retrain models with new version
- Legacy files should be gracefully rejected with clear messaging

## Labels
`bug`, `error-handling`, `user-experience`, `model-loading`, `tensorflow-deprecation`, `pytorch`, `medium-priority`
